### DISCUSSION TITLE
Machine-ID TLS routing not working anymore until 9.3?

### DISCUSSION THREAD

**Discussion by hexinal** [gravitational/teleport#12759] [Q&A] [answered]
I had a machine-id setup that used to work without a dedicated ssh listener (9.0.4?), after a redeployment I can't seem to make it work now.

```bash
[...]
debug1: Executing proxy command: exec "/usr/bin/tsh" proxy ssh --cluster=teleport.local --proxy=teleport.local user@node.teleport.local:3022
[...]
ssh: connect to host teleport.local port 3023: Connection refused
ERROR: exit status 255
```

Any reasons that it seems "hardcoded" for the default ssh listener port to be used for the connection?

I think what I did at the time was to modify the ssh_config tbot generated to use the standard one of tsh. 

I guess it will be supported explicitly with 9.3 https://github.com/gravitational/teleport/commit/24a7c55ddd17500c195beb68243d3f266270aa9d

EDIT: It does work w/ a dedicated ssh listener, so it's not that problematic, just verifying if I missed something.

**Comment by webvictim** [comment, marked as answer]
TLS routing will be officially supported when 9.3 is released as you noted. Good to know that the dedicated SSH listener workaround still works until then.
